tqa-web-app/src/app/state/store/resources/adminauth/adminauth.reducer.ts

```diff
@@ -0,0 +1,43 @@
+import { Action, ActionCreator, createReducer, Creator, on } from "@ngrx/store";
+import { AdminV2 } from "src/app/dtos/v2/admin-auth.dto.v2";
+import { CRState, initialCRState, ReadableState } from "../resource.state";
+import * as adminAuthActions from "./adminauth.actions";
+import { adminLoginSuccess } from "./adminauth.actions";
+//import { AdminAuthState, initialAdminState } from "./adminauth.state";
+
+
+export interface AdminAuthState extends ReadableState<AdminV2>{};
+export const initialAdminState: AdminAuthState = {
+    ...initialCRState
+};
+
+
+const onAdminLoginSuccess =     on<AdminAuthState, ActionCreator<string, Creator<any[], object>>[]>(
+    adminAuthActions.adminLoginSuccess,
+    (state: AdminAuthState, props:any): AdminAuthState =>({
+        ...state,
+        resource:props.admin,
+        loading:true
```

Comment:
loading should be false on admin login success, the login process is no longer happening.  Loaded would be true, indicating that the login process has completed.

Reply:
amended. One thing I don't understand, what is these two statuses are used for?

Reply:
I only ended up using the statuses once, but they are to allow UIs to show real time status indicators.  When loading, you can show the user a spinner, once totally loaded you can remove the spinner.   Things like that.  You need some type of state changes to allow for that behavior.  I am using those state options when creating visitors if you want to see an example.